### PR TITLE
Switch tests from java to scala

### PR DESCRIPTION
**Pull request by davebassan**
- Switch java E2E tests to scala and create as functional tests under corresponding module
- Split out integration tests from unit tests where a container or embedded server is used
- Update github actions workflow to cache assemblies and only upload after functional tests stage

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -196,13 +202,14 @@ lazy val jms = (projectMatrix in file("kafka-connect-jms"))
...
-      )
+        dependencyOverrides ++= nettyOverrides,
+      ),
   )
   .kafka2Row()
   .kafka3Row()
   .configureAssembly()
   .configureTests(kafkaConnectJmsTestDeps)
+  .configureIntegrationTests(kafkaConnectJmsTestDeps)
   .disableParallel()
```
Comment (davidsloan) [inline build.sbt:213]: is disableParallel() still necessary for the unit tests?

File: build.sbt [hunk 2]
```diff
@@ -231,13 +238,14 @@ lazy val mqtt = (projectMatrix in file("kafka-connect-mqtt"))
...
+        dependencyOverrides ++= nettyOverrides,
+      ),
   )
   .kafka2Row()
   .kafka3Row()
   .configureAssembly()
-  .configureTests(kafkaConnectMqttTestDeps)
+  .configureTests(baseTestDeps)
+  .configureIntegrationTests(kafkaConnectMqttTestDeps)
   .disableParallel()
```
Comment (davidsloan) [inline build.sbt:249]: Is disableParallel still necessary for the unit tests?

File: kafka-connect-aws-s3/src/it/scala/io/lenses/streamreactor/connect/aws/s3/SlowTest.scala [hunk 3]
```diff
@@ -1,5 +1,5 @@
 package io.lenses.streamreactor.connect.aws.s3
+
 import org.scalatest.Tag
 
 object SlowTest extends Tag("io.lenses.tags.SlowTest")
```
Comment (davidsloan) [inline kafka-connect-aws-s3/src/it/scala/io/lenses/streamreactor/connect/aws/s3/SlowTest.scala:5]: Did you mention that these tags can be deleted now?

Comment (davebassan) [reply kafka-connect-aws-s3/src/it/scala/io/lenses/streamreactor/connect/aws/s3/SlowTest.scala:5]: Missed that. Thanks. I will remove them.

**Pull request merged**